### DISCUSSION TITLE
Media Browser Image Thumbnails

### DISCUSSION THREAD

**Discussion by danbayliss** [home-assistant/feature-requests#480] [Integration enhancements]
I use the media browser to show recent images of my camera when a person has been detected. It would be good if this showed thumbnails and also let you use the left and right keyboard arrows to scroll through them when you have opened one

Thanks!

**Comment by tbrausch** [comment]
I was just getting ready to put in a request for the same feature when I saw this post.  I use the Media Browser to display camera recordings (.mp4 files) that are created upon motion detection, and I agree it would be nice to see a thumbnail to possibly see what caused the motion event .  Additional sort options would also be a plus.  I used to use the Gallery Card, and if you have not tried it, I suggest you do so.  If your files are hosted locally this custom lovelace card works perfectly.

`https://github.com/TarheelGrad1998/gallery-card`

However, I have recently converted to saving my video files to an SMB share (for size considerations), and the card will no longer display the remote files.  It would be fantastic to upgrade the Media Browser to provide the same type of funtionality as the Gallery Card.

Note: I did not mention this before, but the Media Browser has no problems seeing and playing my video files that are mounted from an SMB share.

**Reply by tbrausch** [reply]
SOLVED:  I was able to get the Gallery Card working with recordings on an SMB share, so this is a viable solution.

**Reply by danbayliss** [reply]
Thanks! I'll give it a try but it would be nice to see the inbuilt browser to have this functionality

**Reply by cby016** [reply]
I tried using the Gallery Card but sometimes my video clips hang up and don't play in the gallery card. They play much quicker in the media browser.  It would be nice to see a thumbnail of the video clip instead of a camera icon.

**Reply by ildar170975** [reply]
It would be great if this "show thumbnails" functionality is implemented as a native HA feature.
Also found at least one more discussion https://github.com/home-assistant/frontend/discussions/11312 (not to mention my https://github.com/home-assistant/frontend/discussions/17955).
Sad that there is no any feedback from Dev team.

**Comment by FalconUK** [comment]
Have been looking for the most suitable/most recent discussion to add my comments to, seems this may be the one.

Media browser doesn't:
- show previews for images
- doesn't provide 'select all', only 'deselect all', when it comes to deleting fines.

Both of these would be useful features.

I'm using the HACS gallery view to preview the images, but that doesn't have select/deselect/delete functionality.

**Comment by Neiphos** [comment]
Sorry to bump an old thread but i assume this is still an issue? i have just come across the same problem having .jpg images in the media folder and not having a thumbnail available, but the images appear when clicked on.

**Reply by xavirii** [reply]
Yes same issue here

**Comment by karwosts** [comment]
Looked at this a bit and this is what I've found. 

- This isn't a frontend feature but a core feature. The media browser UI already fully supports showing thumbnails for items, but the integration that provides the media directory needs to provide a path to a thumbnail image file.
- For the case of generic local media files or mounted paths mapped via `media_dirs`, that integration is `media_source`. And indeed looking at that integration I confirmed it does not provide any thumbnail images. 
- A years ago an [old PR](https://github.com/home-assistant/core/pull/49133/) was created to have `media_source` automatically generate and cache thumbnail images for items. It was declined with Paulus saying he wasn't sure that he wanted home assistant to be responsible for generating and maintaining these files.
- Paulus suggested that maybe we could just work with pre-existing thumbnails, if they existed. Even if we did support that, it would put the onus on the user to create their own thumbnail images. And there is no universal standard for how those files would be formatted. I started an architectural [discussion](https://github.com/home-assistant/architecture/discussions/1263) to see if we could agree on a standard format, so users could generate these if they wanted to, and then they could be used. That may or may not ever lead to anything though.


(Moved to main feature requests repo as it is not a frontend feature)

**Comment by sillvva** [comment]
Hoping this gets added soon. It would be nice to see a gallery of my camera snapshots.
<img width="1862" height="863" alt="CleanShot 2026-05-11 at 17 58 17" src="https://github.com/user-attachments/assets/2b489088-fbaf-44b7-80de-4622c2a824a6" />
